### PR TITLE
Summary metrics for AUC, FNR, TNR.

### PR DESCRIPTION
**Pull request by lucyvasserman**

### CODE REVIEW

File: src/model_bias_analysis.py [hunk 1]
```diff
@@ -255,6 +256,43 @@ def per_term_negative_rates(df, terms, model_families, threshold, text_col,
...
     return pd.DataFrame(records)
 
+### Summary metrics
+def diff_from_base(base_metrics, per_term_results, model_families, metric_column):
+    diffs = {}
+    for fams in model_families:
+        family_name = model_family_name(fams)
+        family_base_metrics = base_metrics[family_name]
+        metric_diff_sum = 0.0
+        for metric_list in per_term_results[family_name + metric_column]:
```
Comment (jetpack) [inline src/model_bias_analysis.py]: I find the nested for loops confusing (despite coming up with the structure of these dataframes :-/).

I think this is iterating over the per-term rows, right? Maybe add a comment saying this loop is over each row, and that the 'metric_list' contains values of the metric for each copy of a model within the same model family.

File: src/model_bias_analysis.py [hunk 2]
```diff
@@ -255,6 +256,43 @@ def per_term_negative_rates(df, terms, model_families, threshold, text_col,
...
+            for i in range(10):
+                metric_diff_sum += abs(family_base_metrics[i] - metric_list[i])
```
Comment (jetpack) [inline src/model_bias_analysis.py]: Let's not hardcode 10. Something like this instead?

```
per_model_metric_diffs = [abs(overall_score - per_term_score)
                                          for overall_score, per_term_score
                                          in zip(family_base_metrics, metric_list)]
diffs[family_name] = sum(per_model_metric_diffs)

```

File: src/model_bias_analysis.py [hunk 3]
```diff
@@ -226,8 +224,11 @@ def per_term_negative_rates(df, terms, model_families, threshold, text_col,
     """
     records = []
     for term in terms:
-        term_subset = df[df[text_col].str.contains(r'\b{}\b'.format(term),
-                                                   case=False)]
+        if term != 'ALL':
```
[the remark this answers is not in the document]
Comment (lucyvasserman) [reply src/model_bias_analysis.py]: Done

File: src/model_bias_analysis.py [hunk 4]
```diff
@@ -255,6 +256,43 @@ def per_term_negative_rates(df, terms, model_families, threshold, text_col,
         records.append(term_record)
     return pd.DataFrame(records)
 
+### Summary metrics
+def diff_from_base(base_metrics, per_term_results, model_families, metric_column):
```
[the remark this answers is not in the document]
Comment (lucyvasserman) [reply src/model_bias_analysis.py]: Done

File: src/model_bias_analysis.py (hunk 1 above)
Comment (lucyvasserman) [reply src/model_bias_analysis.py]: Done

File: src/model_bias_analysis.py (hunk 2 above)
Comment (lucyvasserman) [reply src/model_bias_analysis.py]: Good point, this is much nicer.

**Comment by jetpack** [comment]
:+1: feel free to hit the pretty green "Merge pull request" button to merge your changes into the master branch.

**Pull request merged**